tests/lib/ast-utils.js

```diff
@@ -904,4 +904,38 @@ describe("ast-utils", () => {
             });
         });
     });
+
+    describe("couldBeError", () => {
+        const EXPECTED_RESULTS = {
+            5: false,
+            null: false,
+            true: false,
+            "'foo'": false,
+            "`foo`": false,
+            foo: true,
+            "new Foo": true,
+            "Foo()": true,
+            "foo`bar`": true,
+            "foo.bar": true,
+            "(foo = bar)": true,
+            "(foo = 1)": false,
+            "(1, 2, 3)": false,
+            "(foo, 2, 3)": false,
+            "(1, 2, foo)": true,
+            "1 && 2": false,
+            "1 && foo": true,
+            "foo && 2": true,
+            "foo ? 1 : 2": false,
+            "foo ? bar : 2": true,
+            "foo ? 1 : bar": true
```

Comment:
I would be fine with that behavior, but I'm wondering if it would cause confusion because the rule is called `no-reject-literal` and object expressions are not literals.

Also, I think that would be a breaking change for `no-throw-literal`, (unless we explicitly made `no-throw-literal` more lenient by allowing `ObjectExpressions`).